cloud-arm/src/main/java/com/sequenceiq/cloudbreak/cloud/arm/ArmSetup.java

```diff
@@ -43,6 +52,56 @@
     private SyncPollingScheduler<Boolean> syncPollingScheduler;
     @Inject
     private ArmTemplateUtils armTemplateUtils;
+    @Inject
+    private ResourceNotifier resourceNotifier;
+
+    @Override
+    public void prepareImage(AuthenticatedContext authenticatedContext, CloudStack stack) {
+        String osStorageName = armClient.getStorageName(authenticatedContext.getCloudContext());
+        String storageGroup = armTemplateUtils.getStackName(authenticatedContext.getCloudContext());
+        AzureRMClient client = armClient.createAccess(authenticatedContext.getCloudCredential());
+        try {
+            if (!storageContainsImage(client, storageGroup, osStorageName, stack.getImage().getImageName())) {
+                client.copyImageBlobInStorageContainer(storageGroup, osStorageName, IMAGES, stack.getImage().getImageName());
+            }
+        } catch (HttpResponseException ex) {
+            throw new CloudConnectorException(ex.getResponse().getData().toString(), ex);
+        } catch (Exception ex) {
+            throw new CloudConnectorException(ex);
+        }
+        LOGGER.debug("prepare image has been executed");
+    }
+
+    @Override
+    public CheckImageProgress checkImageStatus(AuthenticatedContext authenticatedContext, CloudStack stack) {
+        String osStorageName = armClient.getStorageName(authenticatedContext.getCloudContext());
+        String storageGroup = armTemplateUtils.getStackName(authenticatedContext.getCloudContext());
+        ArmCredentialView armCredentialView = new ArmCredentialView(authenticatedContext.getCloudCredential());
+        AzureRMClient client = armClient.createAccess(armCredentialView);
+        try {
+
+            CopyState copyState = client.getCopyStatus(storageGroup, osStorageName, IMAGES, stack.getImage().getImageName());
+            if (CopyStatus.SUCCESS.equals(copyState.getStatus())) {
+                return new CheckImageProgress(ImageStatus.CREATE_FINISHED, ONE_HUNDRED);
+            } else if (CopyStatus.ABORTED.equals(copyState.getStatus()) || CopyStatus.INVALID.equals(copyState.getStatus())) {
+                return new CheckImageProgress(ImageStatus.CREATE_FAILED, 0);
+            } else {
+                LOGGER.info(String.format("CopyStatus Pending %s byte/%s byte: %.4s %%",
+                        copyState.getTotalBytes(), copyState.getBytesCopied(), ((double) copyState.getBytesCopied() / (double) copyState.getTotalBytes())
```

Comment:
I would use just inrteger operations, e.g. multiple by 100 and after that divide it